components/blitz/src/omero/gateway/facility/BrowseFacility.java

```diff
@@ -269,6 +315,50 @@ public IObject findIObject(SecurityContext ctx, String klassName, long id,
         }
         return null;
     }
+    
+    /**
+     * Retrieves an updated version of the specified object.
+     *
+     * @param ctx
+     *            The security context.
+     * @param pojoName
+     *            The type of object to retrieve. (Either the simple or the full
+     *            class name, e. g. omero.gateway.model.DatasetData or
+     *            DatasetData)
+     * @param id
+     *            The object's id.
+     * @param allGroups
+     *            Pass <code>true</code> to look for all groups
+     * @return The last version of the object.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public DataObject findObject(SecurityContext ctx, String pojoName, long id,
+            boolean allGroups) throws DSOutOfServiceException,
```

Comment:
You mean for example one method `getImages(SecurityContext ctx, Options options)` instead of the - at the moment - 7 different `getImages()` methods? That would make sense/simplify the API. Should have done that from the start; now it means I'd have to mark most of the methods as deprecated. That'd be a change for the next round anyway, too much for the 5.2.2.